src/algebra/group_power/lemmas.lean

```diff
@@ -335,19 +335,23 @@ end linear_ordered_add_comm_group
   ((n • a : A) : with_bot A) = n • a :=
 add_monoid_hom.map_nsmul ⟨(coe : A → with_bot A), with_bot.coe_zero, with_bot.coe_add⟩ a n
 
+lemma mul_nsmul_left [non_unital_non_assoc_semiring R] (a b : R) :
+  ∀ n : ℕ, n • (a * b) = a * n • b
+| 0 := by simp_rw [zero_nsmul, mul_zero]
+| (n + 1) := by simp_rw [succ_nsmul, mul_nsmul_left, mul_add]
+
+lemma mul_nsmul_assoc [non_unital_non_assoc_semiring R] (a b : R) :
+  ∀ n : ℕ, n • (a * b) = n • a * b
+| 0 := by simp_rw [zero_nsmul, zero_mul]
+| (n + 1) := by simp_rw [succ_nsmul, mul_nsmul_assoc, add_mul]
+
```

Comment:
I just created #13002 to remove these lemmas, I think we have more general versions